core/src/replay_stage.rs

```diff
@@ -578,6 +580,20 @@ impl ReplayStage {
             let _exit = Finalizer::new(exit.clone());
             let mut identity_keypair = cluster_info.keypair().clone();
             let mut my_pubkey = identity_keypair.pubkey();
+            if my_pubkey != startup_identity {
+                // set-identity was called during the startup procedure, ensure the tower is consistent
+                // before starting the loop. further calls to set-identity will reload the tower in the loop
+                tower = Self::reload_tower_with_new_identity(
+                    &tower_storage,
+                    &my_pubkey,
+                    &vote_account,
+                    &bank_forks,
+                );
+                warn!(
+                    "Identity changed from {} to {}",
+                    startup_identity, my_pubkey
+                );
+            }
```

Comment:
Why do we need a warn! here but not in the loop?

Reply:
we do warn in loop see line 1011

Reply:
Ok, if you always warn you can put that in the common function I guess

Reply:
would prefer not to, as the helper is just a generic load of the tower. it doesn't need to know anything about the previous pubkey.